packages/android_alarm_manager/test/android_alarm_manager_test.dart

```diff
@@ -0,0 +1,203 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:android_alarm_manager/android_alarm_manager.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  setUpAll(() {
+    AndroidAlarmManager.channel
+        .setMockMethodCallHandler((MethodCall call) => null);
+  });
+
+  test('${AndroidAlarmManager.initialize}', () async {
+    AndroidAlarmManager.channel
+        .setMockMethodCallHandler((MethodCall call) async {
+      assert(call.method == 'AlarmService.start');
+      return true;
+    });
+
+    final bool initialized = await AndroidAlarmManager.initialize();
+
+    expect(initialized, isTrue);
+  });
+
+  group('${AndroidAlarmManager.oneShotAt}', () {
+    test('validates input', () async {
+      final DateTime validTime = DateTime.utc(1993);
+      final int validId = 1;
+      final Function validCallback = (int id) => null;
+
+      // Callback should take a single int param.
+      expect(
+          () => AndroidAlarmManager.oneShotAt(
+              validTime, validId, ((String foo) => foo)),
+          throwsAssertionError);
+
+      // ID should be less than 32 bits.
+      expect(
+          () => AndroidAlarmManager.oneShotAt(
+              validTime, 2147483648, validCallback),
+          throwsAssertionError);
+    });
+
+    test('sends arguments to the platform', () async {
+      final DateTime alarm = DateTime(1993);
+      const int rawHandle = 4;
+      InstanceWrappers.setMockCallbackHandleFn(
+          (Function _) => CallbackHandle.fromRawHandle(rawHandle));
+
+      final int id = 1;
+      final bool alarmClock = true;
+      final bool allowWhileIdle = true;
+      final bool exact = true;
+      final bool wakeup = true;
+      final bool rescheduleOnReboot = true;
+
+      AndroidAlarmManager.channel
+          .setMockMethodCallHandler((MethodCall call) async {
+        expect(call.method, 'Alarm.oneShotAt');
+        expect(call.arguments[0], id);
+        expect(call.arguments[1], alarmClock);
+        expect(call.arguments[2], allowWhileIdle);
+        expect(call.arguments[3], exact);
+        expect(call.arguments[4], wakeup);
+        expect(call.arguments[5], alarm.millisecondsSinceEpoch);
+        expect(call.arguments[6], rescheduleOnReboot);
+        expect(call.arguments[7], rawHandle);
+        return true;
+      });
+
+      final bool result = await AndroidAlarmManager.oneShotAt(
+          alarm, id, (int id) => id,
```

Comment:
Same as above: avoid using anonymous closures in the tests as they're not supported in general. There's a few more instances where this is done below.